A35-grpc-gateway.md

```diff
@@ -0,0 +1,83 @@
+Integrate the grpc-gateway as a multi-language option to allow for REST endpoint support
+----
+* Author(s): Keith Adler (Talador12)
+* Approver: a11r
+* Status: Draft --- {Draft, In Review, Ready for Implementation, Implemented}
+* Implemented in: Core (C++), Java, Python, and Go.
```

Comment:
There are two big chunks to this feature: codegen and the runtime HTTP/1 server. Codegen is definitely language-specific, but the runtime aspect doesn't have to be. If possible, the runtime aspect should be implemented entirely in Core, so that it won't have to be *re*implemented for Python, Ruby, PHP, and C#.

Is there any particular reason why you called Python out as the only wrapped language here?

I don't think it's an absolute must that this feature be implemented in all languages from the get-go, but I think we should design it in such a way that adding support to additional languages in the future is as easy as possible.

Reply:
I agree on Core first. I called out Python first because I am working primarily in python projects/libraries and think it is a decent option.

```
I don't think it's an absolute must that this feature be implemented in all languages from the get-go, but I think we should design it in such a way that adding support to additional languages in the future is as easy as possible.
```
Agreed. Better to do this the right way than a rush job.

I'll add more detail about those 2 facets.